src/bin/ion/commands/generate/templates/java/class.templ

```diff
@@ -39,6 +39,45 @@ import java.io.IOException;
         }
     {% endfor %}
 
+    public static class {{ model.name }}Builder {
+        {% for field_name, field_val in struct_info["fields"] -%}
+            {% set field_value = field_val.0 | fully_qualified_type_name %}
+            private java.util.Optional<{{ field_value | wrapper_class }}> {{ field_name | camel }};
```

Comment:
For the builder in particular, I don't think it makes sense to use `Optional` at all. You start by setting them all to `Optional.empty()`, and then you set the values to `Optional.ofNullable(...)`, which when the value is `null` produces `Optional.empty()`. Finally, when you construct the instance, you have `someOptional.orElse(null)`.

In the builder, `Optional.empty()` is just an expensive `null`.

Reply:
I had used `Optional` here since required field were still using primitive types instead of a wrapper class. Using `Optional` made it easier to verify if the required fields were set or not without using any checks for default values for primitive types. 

I will take a look at using wrappers inside the builder for all fields(required/optional) and remove `Optional` usage here.